src/rpc/mining.cpp

```diff
@@ -472,17 +472,17 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
         // Release the wallet and main lock while waiting
         LEAVE_CRITICAL_SECTION(cs_main);
         {
-            checktxtime = boost::get_system_time() + boost::posix_time::minutes(1);
+            checktxtime = std::chrono::steady_clock::now() + std::chrono::minutes(1);
 
-            boost::unique_lock<boost::mutex> lock(csBestBlock);
+            CWaitableLock lock(csBestBlock);
             while (chainActive.Tip()->GetBlockHash() == hashWatchedChain && IsRPCRunning())
             {
-                if (!cvBlockChange.timed_wait(lock, checktxtime))
+                if (cvBlockChange.wait_until(lock, checktxtime) == std::cv_status::timeout)
```

Comment:
cvBlockChange needs to be notified in Interrupt() in init, otherwise we'll hang here at shutdown.

Reply:
@theuni Something like this? :-)

```diff
diff --git a/src/init.cpp b/src/init.cpp
index 25b8caf..73f4860 100644
--- a/src/init.cpp
+++ b/src/init.cpp
@@ -167,6 +167,7 @@ void Interrupt(boost::thread_group& threadGroup)
     if (g_connman)
         g_connman->Interrupt();
     threadGroup.interrupt_all();
+    cvBlockChange.notify_all();
 }

 void Shutdown()
```

Reply:
Looking deeper, I think this is actually ok (though circuitous) as-is.
```
StopRPC() -> g_rpcSignals.Stopped() -> OnRPCStopped() -> cvBlockChange.notify_all()
```
So, nevermind my comment.